### PR TITLE
Update MPS to 3.2.1

### PR DESCRIPTION
**Pull request by gord1anknot**
Added zap and caveat stanzas found on casks for other intellij
based IDEs.

### CODE REVIEW

**Comment by fanquake** [comment]
@radeksimko Is there anything holding this up from being merged?

**Comment by radeksimko** [comment]
Sorry for being unresponsive here.

It's just necessary to double check that the caveat is actually valid. That workaround works well for the original IDEA, but I'm not convinced that it works for other editors like MPS, RubyMine, PyCharm, etc.

The first part about Java 6 will likely be true, but it's worth double checking that this is true:

```
To use existing newer Java at your own risk,
    add JVMVersion=1.6+ to ~/Library/Preferences/MPS#{version.tr('.','')[0,2]}/idea.properties
```

**Comment by gord1anknot** [comment]
Since it's my PR I've checked that assumption. I found that it's not true on this Mountain Lion macbook pro machine - but in a way that would make it not true for any non-Intellij IDE in all likelihood, not just mps. I was able to reproduce the intended result, i.e.; running on Oracle Java installed by homebrew-cask, by editing the Info.plist with sed. I'm happy to update the PR with that methodology instead. Methodology as follows:

- install mps from my pr: `brew cask install https://raw.githubusercontent.com/gord1anknot/homebrew-cask/13d65776cb0207bd3aa392a0fcb695388e9f05e7/Casks/mps.rb`

- open mps: `open /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS\ 3.2.app`

- check the running mps process to see how it's running. Notice that there's a binary but no java process; Maybe it's just dynamically linked or something. I'm not a desktop developer ;)

```sh
$ ps -awx | grep mps
94028 ??         0:19.94 /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS 3.2.app/Contents/MacOS/mps -psn_0_12794931
94593 ??         0:00.02 /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS 3.2.app/Contents/bin/fsnotifier
```

- Next I applied the config as suggested in the caveat:

```
trash ~/Library/Preferences/MPS32/idea.properties ; echo "JVMVersion=1.6+" >> ~/Library/Preferences/MPS32/idea.properties
pkill mps
```

- Then I opened MPS again, and checked the running mps process, and see almost the same thing:

```sh
$ ps -awx | grep mps
94028 ??         0:19.94 /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS 3.2.app/Contents/MacOS/mps -psn_0_12794931
94593 ??         0:00.02 /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS 3.2.app/Contents/bin/fsnotifier
```

- To ensure apple java is NOT being used, move it out of the way, and then restarted MPS again. I see a crash:

```sh
$ sudo mv /System/Library/Java/JavaVirtualMachines/1.6.0.jdk/ /System/Library/Java/JavaVirtualMachines/1.6.0.jdk.bak
$ open /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS\ 3.2.app/

LSOpenURLsWithRole() failed with error -10810 for the file /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS 3.2.app.
```

Taking a cue from [this project](https://github.com/wonder-mice/mac-java-launcher/blob/master/use) I edit the `Info.plist`:

```sh
sed -i.bak -e '/<string>/ s/1.6\*/1.6+/' /opt/homebrew-cask/Caskroom/mps/3.2.1/MPS\ 3.2.app/Contents/Info.plist
```

And voila:

jetbrains_mps

**Comment by radeksimko** [comment]
@gord1anknot That `Info.plist` hack will work on first look for all JetBrains IDEs, but it's not recommended by vendor for various reasons, see https://github.com/caskroom/homebrew-cask/pull/9943

So in that sense, if you can update the PR to remove the last two lines from `caveats` section, I'm happy to give a it final review & merge it.

**Comment by gord1anknot** [comment]
Great! I've done as you suggested.

File: Casks/mps.rb [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
 
-  url "http://download-cf.jetbrains.com/mps/#{version.tr('.','')}/MPS-#{version}-macos.dmg"
+  url "http://download-cf.jetbrains.com/mps/#{version.tr('.','')[0,2]}/MPS-#{version}-macos.dmg"
   name 'MPS'
   name 'JetBrains MPS'
   homepage 'https://www.jetbrains.com/mps'
   license :apache
 
-  app "MPS #{version}.app"
+  app "MPS #{version[0,3]}.app"
```
Comment (radeksimko) [inline Casks/mps.rb]: The `[0-3]` feels a bit superfluous here, but it's no harm...

**Comment by radeksimko** [comment]
Thanks for tuning the PR and sorry to keep you waiting!

**Pull request merged**